src/Microsoft.OData.Core/Json/DefaultJsonWriterFactory.cs

```diff
@@ -37,11 +38,13 @@ public DefaultJsonWriterFactory(ODataStringEscapeOption stringEscapeOption)
         /// <summary>
         /// Creates a new JSON writer of <see cref="IJsonWriter"/>.
         /// </summary>
-        /// <param name="textWriter">Writer to which text needs to be written.</param>
+        /// <param name="stream">The stream write to.</param>
         /// <param name="isIeee754Compatible">True if it is IEEE754Compatible.</param>
         /// <returns>The JSON writer created.</returns>
-        public IJsonWriter CreateJsonWriter(TextWriter textWriter, bool isIeee754Compatible)
+        public IJsonWriter CreateJsonWriter(Stream stream, bool isIeee754Compatible, Encoding encoding)

```

Comment:
This seems like a breaking change. Why not have two constructors, one for steams and one for text and in the underlying implementation convert the text into a stream ?

Reply:
@wachugamaina It is a breaking change. ODL 8.x will be a major release and it affords us the opportunity to roll out breaking changes that we can't safely make in ODL 7